gslib/tests/test_storage_url.py

```diff
@@ -135,3 +136,38 @@ def test_urls_are_mix_of_objects_and_buckets_is_true_for_a_mix(self):
   def test_urls_are_mix_of_objects_and_buckets_is_null_for_invalid(self):
     urls = list(map(storage_url.StorageUrlFromString, ['gs://b', 'f:o@o:o']))
     self.assertIsNone(storage_url.UrlsAreMixOfBucketsAndObjects(urls))
+
+  def test_urls_raise_error_if_mix_of_objects_and_buckets(self):
+    urls = list(map(storage_url.StorageUrlFromString, ['gs://b1', 'gs://b/o']))
+    with self.assertRaisesRegex(
+            CommandException,
+            'Cannot operate on a mix of buckets and objects.'):
+      storage_url.RaiseErrorIfUrlsAreMixOfBucketsAndObjects(urls, recursion_requested=False)
+
+  def test_urls_raise_error_if_objects_and_buckets_are_mixed_different(self):
```

Comment:
Instead of "_different", you can call the first test "bucket_followed_by_object" and the second "object_followed_by_bucket"